src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -1069,59 +1069,24 @@ private void Save()
             ExecuteSynchronized(() => FileSystemUtility.AddFile(ConfigFilePath, ConfigXDocument.Save));
         }
 
-#if IS_CORECLR
         private static Mutex _globalMutex = new Mutex(initiallyOwned: false);
 
+#if IS_CORECLR
+
         /// <summary>
         /// Wrap all IO operations on setting files with this function to avoid file-in-use errors
         /// </summary>
         private void ExecuteSynchronized(Action ioOperation)
         {
             if (RuntimeEnvironmentHelper.IsWindows)
             {
-                ExecuteSynchronizedCore(ioOperation);
+                ExecuteSynchronizedCoreWindows(ioOperation);
                 return;
             }
             else
             {
-                // Cross-plat CoreCLR doesn't support named lock, so we fall back to
-                // process-local synchronization in this case
-                var owner = false;
-                try
-                {
-                    owner = _globalMutex.WaitOne(TimeSpan.FromMinutes(1));
-                    ioOperation();
-                }
-                catch (InvalidOperationException e)
-                {
-                    throw new NuGetConfigurationException(
-                        string.Format(CultureInfo.CurrentCulture,Resources.ShowError_ConfigInvalidOperation, ConfigFilePath, e.Message), e);
-                }
-
-                catch (UnauthorizedAccessException e)
-                {
-                    throw new NuGetConfigurationException(
-                        string.Format(CultureInfo.CurrentCulture,Resources.ShowError_ConfigUnauthorizedAccess, ConfigFilePath, e.Message), e);
-                }
-
-                catch (XmlException e)
-                {
-                    throw new NuGetConfigurationException(
-                        string.Format(CultureInfo.CurrentCulture,Resources.ShowError_ConfigInvalidXml, ConfigFilePath, e.Message), e);
-                }
-
-                catch (Exception e)
-                {
-                    throw new NuGetConfigurationException(
-                        string.Format(CultureInfo.CurrentCulture,Resources.Unknown_Config_Exception, ConfigFilePath, e.Message), e);
-                }
-                finally
-                {
-                    if (owner)
-                    {
-                        _globalMutex.ReleaseMutex();
-                    }
-                }
+                ExecuteSynchronizedCoreCrossPlat(ioOperation);
+                return;
```

Comment:
Given that both cases in this if-else end in return, the return can be moved out one block.